code/modules/hydroponics/hydroponics_chemreact.dm

```diff
@@ -0,0 +1,287 @@
+  * stat changes as opposed to 271 lines of individual statline effects. Shoutout to the original comments on chems, I just cleaned a few up.
+  */
+/obj/machinery/hydroponics/proc/applyChemicals(datum/reagents/S, mob/user)
+	///Checks if the tray has enough of the mutation chems, otherwise skips that portion.
+	var/has_mutation_chem = FALSE
+	///Checks if the tray has regular or advanced mutation chems, otherwise skips that portion.
+	var/has_nutrients = FALSE
+	///List containing all the nutrient chems that affect plants.
+	var/static/list/nutrients = list(/datum/reagent/plantnutriment/eznutriment,
+						/datum/reagent/plantnutriment/left4zednutriment,
+						/datum/reagent/plantnutriment/robustharvestnutriment,
+						/datum/reagent/medicine/C2/multiver,
+						/datum/reagent/medicine/cryoxadone,
+						/datum/reagent/ammonia,
+						/datum/reagent/saltpetre,
+						/datum/reagent/ash,
+						/datum/reagent/diethylamine,
+						/datum/reagent/consumable/nutriment)
+	///Checks if the tray has any damaging chems, otherwise skips that portion.
+	var/has_toxins = FALSE
+	///list of all the toxic chems that affect plants.
+	var/static/list/toxins = list(/datum/reagent/toxin,
+						/datum/reagent/fluorine,
+						/datum/reagent/chlorine,
+						/datum/reagent/phosphorus,
+						/datum/reagent/consumable/sugar,
+						/datum/reagent/toxin/acid,
+						/datum/reagent/toxin/acid/fluacid,
+						/datum/reagent/toxin/plantbgone,
+						/datum/reagent/napalm,
+						/datum/reagent/toxin/plantbgone/weedkiller,
+						/datum/reagent/toxin/pestkiller,
+						/datum/reagent/drug/nicotine,
+						/datum/reagent/consumable/virus_food)
+	///Checks if the tray has any of the miscellanious or unusual chems, otherwise skips that portion.
+	var/has_other = FALSE
+	var/static/list/others = list(/datum/reagent/consumable/milk,
+						/datum/reagent/consumable/ethanol/beer,
+						/datum/reagent/consumable/honey,
+						/datum/reagent/water/holywater,
+						/datum/reagent/consumable/sodawater,
+						/datum/reagent/stabilizing_agent,
+						/datum/reagent/blood,
+						/datum/reagent/medicine/strange_reagent,
+						/datum/reagent/medicine/adminordrazine)
+	if(myseed)
+		myseed.on_chem_reaction(S) //In case seeds have some special interactions with special chems, currently only used by vines
+
+	//Alright, for optimization sake these checks look for if the tray has reagents of a specific type, to keep hydroponics trays from checking everything, every proc.
+	if(S.has_reagent(/datum/reagent/toxin/mutagen, 0.5) || S.has_reagent(/datum/reagent/uranium/radium, 1) || S.has_reagent(/datum/reagent/uranium, 1))
+		has_mutation_chem = TRUE
+	for(var/datum/reagent/i in nutrients)
+		if(S.has_reagent(i))
+			has_nutrients = TRUE
+			break
+	for(var/datum/reagent/i in toxins)
+		if(S.has_reagent(i))
+			has_toxins = TRUE
+			break
+	for(var/datum/reagent/i in others)
+		if(S.has_reagent(i))
+			has_other = TRUE
+			break
```

Comment:
Instead of checking for every existent chem you can simply check the chems inside of the machinery and make them take the desired effect. You create a reagent proc and redefine it as needed.
Something like this:
```nix
for(var/c in S.reagent_list)
	var/datum/reagent/chem = c
	chem.on_hydroponics_application(src)
```